### DISCUSSION TITLE
Unable to issue a certificate with a new root CA

### DISCUSSION THREAD

**Discussion by tmontney** [Keyfactor/ejbca-ce#732] [Q&A] [answered]
New to EJBCA, standing up a fresh docker instance. I'm trying to replace the stock root CA with my own and replace the WildFly admin cert. When creating the CA, I created a new crypto token and replicated the settings from the stock CA (aside from the subject DN and CRL validation data).

When making a new request for a "Server" subtype and selecting my new root CA, I change the key algorithm and set a common name. At the bottom, the download buttons are greyed out. If I hit "Reset", I get "An unexpected problem has occurred: badly formatted directory string". If I try the same values but targeting the stock CA, download buttons work normally.

**Comment by primetomas** [comment]
Can you provide a screenshot of this?

**Comment by tmontney** [comment]
Here's my attempt to issue a test CA server cert from my test CA:

![Issue-TestCA-Cert-01](https://github.com/user-attachments/assets/4c70dd50-dc48-4083-bbc0-8fa704a72990)
![Issue-TestCA-Cert-02](https://github.com/user-attachments/assets/74553c64-a401-4bc8-b5b8-aebd8dc92604)

Buttons at the bottom are disabled. When I hit "reset", I get this:

![Error-When-Resetting](https://github.com/user-attachments/assets/0924196a-d34e-4ec3-8be3-1ab8211dbc25)

Tailing the docker container gives me this stack trace: https://pastebin.com/2Ni5T7wv

Here's my crypto token:

![CryptoToken](https://github.com/user-attachments/assets/7b12d7e6-ea61-4894-930a-8fe2123874df)

And here's my test CA:

![TestCA-01](https://github.com/user-attachments/assets/3c3d5d72-239c-4291-8d62-bbccf430f289)
![TestCA-02](https://github.com/user-attachments/assets/fe18ab06-d0ed-44da-adb8-3b40552623f3)
![TestCA-03](https://github.com/user-attachments/assets/b7c62751-abaa-4856-bb51-3d16765f8cb2)

**Comment by tmontney** [comment, marked as answer]
I went back and looked at this tutorial again: https://docs.keyfactor.com/ejbca/latest/tutorial-start-out-with-ejbca-docker-container#id-(9.1.1)Tutorial-StartoutwithEJBCADockercontainer-Step5-IssueSuperAdmincertificate

I'm not supposed to enter a formatted DN, just the value for CN. If I put something like "myserver" instead of "CN=myserver", the cert can be downloaded. Dumb mistake.

A few things to note:

- The UI shouldn't throw an error like that upon hitting "Reset"
- The UI seems to be inconsistent on validation.

   1. Enter "CN=123" for CN, buttons remain disabled
   2. Remove "CN=", buttons are enabled.
   3. Append that value back, buttons remain enabled.

- Buttons remain enabled when leaving the required "Provide User Credentials" fields empty. If you attempt to download the certificate, each field appends additional HTML "Value is required". While this prevents the user from proceeding, it's inconsistent with disabling the buttons. (For example, either the buttons should become disabled, or error text appears in all cases.)

**Comment by primetomas** [comment]
Great you got it working. Thanks for the feedback on inconsistensies. I'll provide that back to development.
